### PR TITLE
VZ-6784: E2E Tests for API Conversion for new changes using v1beta1 client

### PR DESCRIPTION
**Pull request by Abhijithang**
This PR expands dynamic config e2e test coverage for v1beta1 API changes such as ingressNginx, fluentd, and authproxy.

### CODE REVIEW

File: tests/e2e/update/fluentd/fluentd_update.go [hunk 1]
```diff
@@ -81,6 +106,30 @@ func ValidateDaemonset(osURL, osSec, apiSec string, extra ...vzapi.VolumeMount)
 	pkg.Log(pkg.Info, fmt.Sprintf("Fluentd took %v to update", time.Since(start)))
 }
 
+func ValidateDaemonsetV1beta1(osURL, osSec, apiSec string, extra ...v1beta1.VolumeMount) {
+	start := time.Now()
+	gomega.Eventually(func() bool {
```
Comment (desagar) [inline tests/e2e/update/fluentd/fluentd_update.go:111]: Eventually should be moved to the top level inside the It block - I can't remember the issue but there were some problems with putting Eventually in a function. The function itself can return a boolean and you can do something like:

gomega.Eventually(ValidateDaemonsetV1beta1, tenMinutes, pollingInterval).Should(gomega.BeTrue(), .....etc)

File: tests/e2e/update/fluentd/fluentd_update_test.go [hunk 2]
```diff
@@ -46,19 +58,46 @@ var _ = t.Describe("Update Fluentd", Label("f:platform-lcm.update"), func() {
 				ElasticsearchSecret: extEsSec + "missing",
 				ElasticsearchURL:    opensearchURL,
 			}}
-			ValidateUpdate(m, "must be created")
+			expectedError := "must be created"
+			gomega.Expect(func() bool {
+				return ValidateUpdate(m, expectedError)
+			}, fmt.Sprintf("expected error %v", expectedError))
```
Comment (desagar) [inline tests/e2e/update/fluentd/fluentd_update_test.go:64]: this and all other calls to ValidateUpdate and ValidateUpdatev1beta1 should have a ".Should" similar to the original Expect call e.g.

.Should(gomega.BeTrue(), fmt.Sprintf("expected error %v", expectedError))

File: tests/e2e/update/fluentd/fluentd_update_test.go [hunk 3]
```diff
@@ -46,19 +58,46 @@ var _ = t.Describe("Update Fluentd", Label("f:platform-lcm.update"), func() {
...
-			ValidateDaemonset(opensearchURL, extEsSec, "")
+			v1beta1Modifier := &FluentdModifierV1beta1{Component: v1beta1.FluentdComponent{
+				OpenSearchSecret: extEsSec,
+				OpenSearchURL:    opensearchURLV1beta1,
+			}}
+
+			// Update CR using v1alpha1 API client.
+			gomega.Expect(func() bool {
+				return ValidateUpdate(v1alpha1Modifier, "")
+			}, fmt.Sprintf("expected error %v", ""))
```
Comment (desagar) [inline tests/e2e/update/fluentd/fluentd_update_test.go:83]: missing .Should(...) - see earlier comment

File: tests/e2e/update/fluentd/fluentd_update_test.go [hunk 4]
```diff
@@ -46,19 +58,46 @@ var _ = t.Describe("Update Fluentd", Label("f:platform-lcm.update"), func() {
...
+			start := time.Now()
+			gomega.Eventually(func() bool {
+				return ValidateDaemonset(opensearchURL, extEsSec, "")
+			}, tenMinutes, pollingInterval).Should(gomega.BeTrue(), fmt.Sprintf("DaemonSet %s is not ready for %v", opensearchURL, time.Since(start)))
+			fmt.Sprintf("Fluentd took %v to update", time.Since(start))
+
+			//Update CR using v1beta1 API client.
+			gomega.Expect(func() bool {
+				return ValidateUpdateV1beta1(v1beta1Modifier, "")
```
Comment (desagar) [inline tests/e2e/update/fluentd/fluentd_update_test.go:92]: missing .Should(...) - see earlier comment

Comment (Abhijithang) [reply tests/e2e/update/fluentd/fluentd_update_test.go:92]: Done

**Review by desagar — APPROVED** [review]

**Review by desagar — APPROVED** [review]

**Pull request merged**